transport/register.go

```diff
@@ -66,22 +67,24 @@ type Registry interface {
 	//
 	// service may be empty to indicate that the default service name should
 	// be used.
-	GetHandler(service, procedure string) (UnaryHandler, error)
+	GetHandlerSpec(service, procedure string) (HandlerSpec, error)
 }
 
 // MapRegistry is a Registry that maintains a map of the registered
 // procedures.
 type MapRegistry struct {
 	defaultService string
-	unaryEntries   map[ServiceProcedure]UnaryHandler
+	entries        map[ServiceProcedure]UnaryHandler
```

Comment:
also, wouldn't you rather use `map[ServiceProcedure]HandlerSpec`? Currently
it's possible to register the same procedure name as oneway and unary.